### PR TITLE
Add to harden-flatpak logic that applies the highest supported hwcap

### PR DESCRIPTION
**Pull request by spaceoden**
checks detected hwcaps support and applies the highest supported hwcap for libhardened_malloc.so

### CODE REVIEW

File: files/system/usr/share/ublue-os/just/60-custom.just [hunk 1]
```diff
@@ -56,10 +56,21 @@ remove-kargs-hardening:
...
     echo "Hardening kargs removed."
 
-# Harden flatpaks by preloading hardened_malloc
+# Harden flatpaks by preloading hardened_malloc (highest supported hwcap)
 harden-flatpak:
     #!/usr/bin/bash
-    flatpak override --user --filesystem=host-os:ro --env=LD_PRELOAD=/var/run/host/usr/lib64/libhardened_malloc.so
+    flatpak override --user --filesystem=host-os:ro
+    #`ld-linux-x86-64.so.2 --help` prints ld.so linker info, including detected hwcaps support. Grep those, then use cut and substring selection (:0:1) to isolate the 1st character after 'v', which will be 4, 3, or 2, corresponding to the system's highest supported hwcap. On x86_64-v1 systems, grep finds no matches, leaving our variables empty.
+    uarches="$(/var/run/host/usr/lib64/ld-linux-x86-64.so.2 --help | grep '(supported, searched)' | cut -d'v' -f2)"
```
Comment (RoyalOughtness) [inline files/system/usr/share/ublue-os/just/60-custom.just]: did you test this? it's broken and has unescaped characters

**Review by RoyalOughtness — CHANGES_REQUESTED** [review]

File: files/system/usr/share/ublue-os/just/60-custom.just (hunk 1 above)
Comment (spaceoden) [reply files/system/usr/share/ublue-os/just/60-custom.just]: Yes, I have tested this. The unescaped ' was in the comment. Github isn't rendering the comment as a comment for some reason.

Comment (RoyalOughtness) [reply files/system/usr/share/ublue-os/just/60-custom.just]: @spaceoden sadly it looks like justfiles might not support inline comments?

Comment (spaceoden) [reply files/system/usr/share/ublue-os/just/60-custom.just]: I moved the # to the 1st character and it's rendering correctly now.

File: files/system/usr/share/ublue-os/just/60-custom.just [hunk 2]
```diff
@@ -56,10 +56,21 @@ remove-kargs-hardening:
...
     echo "Hardening kargs removed."
 
-# Harden flatpaks by preloading hardened_malloc
+# Harden flatpaks by preloading hardened_malloc (highest supported hwcap)
 harden-flatpak:
     #!/usr/bin/bash
-    flatpak override --user --filesystem=host-os:ro --env=LD_PRELOAD=/var/run/host/usr/lib64/libhardened_malloc.so
+    flatpak override --user --filesystem=host-os:ro
+#   `ld-linux-x86-64.so.2 --help` prints ld.so linker info, including detected hwcaps support. Grep those, then use cut and substring selection (:0:1) to isolate the 1st character after 'v', which will be 4, 3, or 2, corresponding to the system's highest supported hwcap. On x86_64-v1 systems, grep finds no matches, leaving our variables empty.
+    uarches="$(/var/run/host/usr/lib64/ld-linux-x86-64.so.2 --help | grep '(supported, searched)' | cut -d'v' -f2)"
```
Comment (RoyalOughtness) [inline files/system/usr/share/ublue-os/just/60-custom.just]: this doesn't work properly:

bash: /var/run/host/usr/lib64/ld-linux-x86-64.so.2: No such file or directory

Comment (RoyalOughtness) [reply files/system/usr/share/ublue-os/just/60-custom.just]: please provide evidence of your testing

Comment (spaceoden) [reply files/system/usr/share/ublue-os/just/60-custom.just]: Interesting. On both of my secureblue installs /var/run/host/usr/lib64/ld-linux-x86-64.so.2 exists. Can you give me more information about your "No such file or directory" error? Does the ld-linux-x86-64.so.2 not exist in that directory on your machine? What about at  /usr/lib64/ ?

Comment (spaceoden) [reply files/system/usr/share/ublue-os/just/60-custom.just]: Here are screenshots of a successful test:

Screenshot from 2024-07-30 06-59-20

Screenshot from 2024-07-30 06-46-00

Screenshot from 2024-07-30 06-46-22

Additionally, this shows /var/run/host/usr/lib64/ld-linux-x86-64.so.2 on my machine:

Screenshot from 2024-07-30 06-42-27

**Review by spaceoden — COMMENTED** [review]
Interesting. On both of my secureblue installs /var/run/host/usr/lib64/ld-linux-x86-64.so.2 exists. Can your search your computer and tell me where ld-linux-x86-64.so.2 can be found on your machine? I can't replicate the problem on mine.

File: files/system/usr/share/ublue-os/just/60-custom.just (hunk 2 above)
Comment (RoyalOughtness) [reply files/system/usr/share/ublue-os/just/60-custom.just]: /var/run/host is only defined for flatpaks I believe. Were you testing from within a flatpak shell? :smile:

Comment (spaceoden) [reply files/system/usr/share/ublue-os/just/60-custom.just]: Not that I'm aware of. But I was using the "dx" edition of bluefin, which apparently adds copies of those libraries in /var/run/host/usr/lib64/ (or makes that directory available to non-flatpak apps). I've now switched to silverblue and see that the libraries are only in /usr/lib64/ . I've updated my fork to point request it from /usr/lib64/ , so the code should now work on non-"dx" images.

Comment (RoyalOughtness) [reply files/system/usr/share/ublue-os/just/60-custom.just]: they probably symlink it

**Pull request merged**